types/express-serve-static-core/index.d.ts

```diff
@@ -440,7 +440,10 @@ interface Request extends http.IncomingMessage, Express.Request {
 
     method: string;
 
-    params: any;
+    params: {
+        [namedParameter: string]: string;
+        [captureGroup: number]: string;
+    };
```

Comment:
But all JavaScript objects only accept strings as keys? Number key lookups would be coerced into strings, e.g. `foo[1] === foo['1']`